src/main/kotlin/org/elm/ide/inspections/MakeDeclarationFromUsageFix.kt

```diff
@@ -0,0 +1,39 @@
+package org.elm.ide.inspections
+
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.elm.lang.core.psi.ElmPsiFactory
+import org.elm.lang.core.psi.elements.ElmFunctionCallExpr
+import org.elm.lang.core.psi.elements.ElmValueExpr
+
+class MakeDeclarationFromUsageFix : NamedQuickFix("Create") {
+    override fun applyFix(element: PsiElement, project: Project) {
+        if (element !is ElmValueExpr) {
+           return
+        }
+
+        // addTopLevelFunction("greet", psiFactory)
+        val psiFactory = ElmPsiFactory(project)
+        val s = element.referenceName
+
+
+        val arity = (element.parent as? ElmFunctionCallExpr)?.arguments?.count() ?: 0
+
+        val argList: String = argListFromArity(arity)
```

Comment:
You can generate type-based parameter names like this:

```kotlin
val inference = element.findInference() ?: return
val callExpr = element.parent as? ElmFunctionCallExpr ?: return
val argsTys = callExpr.arguments.map { inference.elementType(it) }
val parameters = argsTys.joinToString(" ") { it.renderParam() }
```

And then, if `argsTys` isn't empty, you can generate a type annotation the way the [MakeAnnotationIntention does](https://github.com/klazuka/intellij-elm/blob/2d8d9902ba0bcd585fdf0af6251f34160c01c987/src/main/kotlin/org/elm/ide/intentions/MakeAnnotationIntention.kt#L43).

```kotlin
"$s : ${TyFunction(args, TyUnknown()).renderedText(elmFile = element.elmFile).replace("→", "->")}"
```